### PR TITLE
ZoomToLayers method for WMTS and ArcGIS services

### LINKED ISSUE #3819: Implement zoomToLayers method for the rest of the URL types
# Feature Request

Implement zoomToLayers method for the rest of the URL types

### PR DESCRIPTION
**Pull request by FilipLeitner**
## Description

- added zoomToLayers functionality for WMTS and ArcGIS services.  Initial zoom is calculated as combination of extents of each selected layer

## Related issues or pull requests

fixes #3819 

## Pull request type

- [x] Bugfix
- [x] Feature
- [ ] Dependency updates
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe)
- [ ] I am unsure (we'll look into it together)

## Do you introduce a breaking change?

- [ ] Yes
- [x] No
- [ ] I am unsure (no worries, we'll find out)

## Checklist

- [x] I understand and agree that the changes in this PR will be licensed under the [MIT License]
- [ ] I have followed the [guidelines for contributing](https://github.com/hslayers/hslayers-ng/blob/master/CONTRIBUTING.md)
- [ ] The proposed change fits to the content of the [code of conduct](https://github.com/hslayers/hslayers-ng/blob/master/CODE_OF_CONDUCT.md)
- [ ] I have added or updated tests and documentation, and the test suite passes (run `npm test` locally)
- [ ] I'm lost; why do I have to check so many boxes? Please help!

### CODE REVIEW

File: projects/hslayers/src/components/add-data/common/url/layer-table/layer-table.component.html [hunk 1]
```diff
@@ -48,8 +48,15 @@
...
         <table class="table table-sm table-striped table-bordered">
             <thead>
-                <th style="width:1em">&nbsp;</th>
-                <th>{{'COMMON.name' | translateHs: {app} }}</th>
+                <th class="border-0" style="width:1em">&nbsp;</th>
+                <div class="d-flex justify-content-between">
+                    <th class="border-0">{{'COMMON.name' | translateHs: {app} }}</th>
+                    <button [hidden]="type !== 'arcgis' || !data.serviceExpanded" type="button"
+                        class="btn btn-sm px-2 border-0" (click)="collapseServices()"
+                        [title]="'COMMON.stepBack' | translateHs: {app}">
```
Comment (jmacura) [inline projects/hslayers/src/components/add-data/common/url/layer-table/layer-table.component.html:56]: Great idea. I would reword this tooltip to "One level up" ("O úroveň výš") or something like that. What do you think?
Also, why not to put the description right next to the icon? Like 
image
 ?

File: projects/hslayers/src/components/add-data/url/add-data-url.service.ts [hunk 2]
```diff
@@ -7,7 +7,12 @@ import {HsLanguageService} from '../../language/language.service';
 import {HsLayoutService} from '../../layout/layout.service';
 import {HsLogService} from '../../../common/log/log.service';
 import {HsMapService} from '../../map/map.service';
+import {HsToastService} from '../../layout/toast/toast.service';
+
+import {Layer} from 'ol/layer';
+import {Source} from 'ol/source';
 import {transform} from 'ol/proj';
+import {urlDataObject} from './types/data-object.type';
```
Comment (jmacura) [inline projects/hslayers/src/components/add-data/url/add-data-url.service.ts:15]:
```suggestion
import {HsToastService} from '../../layout/toast/toast.service';
import {urlDataObject} from './types/data-object.type';

import {Layer} from 'ol/layer';
import {Source} from 'ol/source';
import {transform} from 'ol/proj';
```

File: projects/hslayers/src/components/add-data/url/types/data-object.type.ts [hunk 3]
```diff
@@ -23,6 +23,7 @@ export type urlDataObject = {
   register_metadata?: boolean;
   resample_warning?: boolean;
   services?: Array<any>;
+  serviceExpanded?: boolean; //Control for ArcGIS table UI
```
Comment (jmacura) [inline projects/hslayers/src/components/add-data/url/types/data-object.type.ts:26]:
```suggestion
  /**
   * Control for ArcGIS table UI
  */
  serviceExpanded?: boolean;
```

**Review by jmacura — COMMENTED** [review]
Please check my comment regarding the "go back" button.

I have tested the functionality of this PR with layers from https://maps.plzensky-kraj.cz/ArcGIS/rest/services (ArcGIS REST) and it seems to zoom as expected 👍

File: projects/hslayers/src/components/add-data/common/url/layer-table/layer-table.component.html (hunk 1 above)
Comment (FilipLeitner) [reply projects/hslayers/src/components/add-data/common/url/layer-table/layer-table.component.html:56]: Sure, it would be more user-friendly. I was thinking of adding some text but didn't come up with anything viable :D. 
I might like 'Back' better just because its shorter ...idk. We can settle this up on telco

**Review by jmacura — APPROVED** [review]
Hey, @FilipLeitner ! I have dared to rebase your commits and hope I have not mess up 🙂 
I have sneaked cs translations into your original commits. If that is fine with you, please merge. Thanks for the improvement!
